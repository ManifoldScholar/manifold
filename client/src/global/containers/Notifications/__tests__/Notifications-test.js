import { NotificationsComponent } from "../";

describe("global/containers/Notifications/Notifications", () => {
  def("notifications", () => ({
    notifications: [
      {
        id: "A_NOTIFICATION_1",
        level: 2,
        heading: "A Heading",
        body: "The body.",
        scope: "global"
      },
      {
        id: "A_NOTIFICATION_2",
        level: 2,
        heading: "A Heading",
        body: "The body.",
        scope: "drawer"
      }
    ]
  }));

  def("root", () => (
    <NotificationsComponent
      notifications={$notifications}
      scope="global"
      dispatch={$dispatch}
    />
  ));

  it("matches the snapshot when rendered", () => {
    expect(mount($withApp($root)).html()).toMatchSnapshot();
  });
});
