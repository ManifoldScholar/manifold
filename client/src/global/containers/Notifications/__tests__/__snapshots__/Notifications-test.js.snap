// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global/containers/Notifications/Notifications matches the snapshot when rendered 1`] = `"<section class=\\"notifications-container\\" role=\\"alert\\"><div class=\\"notifications-list--context-header notifications-list\\"><div><div class=\\"notifications-list__inner\\"><div class=\\"notification notification--error notification--context-header\\"><div class=\\"notification__container\\"><div role=\\"status\\"><p class=\\"notification__heading\\">A Heading</p></div><p class=\\"notification__body\\">The body.</p><button class=\\"notification__button\\" data-id=\\"close\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"manicon-svg notification__button-icon svg-icon--close32\\" width=\\"36\\" height=\\"36\\" viewBox=\\"0 0 32 32\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M7.00007316,7.70847994 L7.70712678,7.00132 L24.9999268,24.2915201 L24.2928732,24.99868 L7.00007316,7.70847994 Z M7.70712678,24.99868 L7.00007316,24.2915201 L24.2928732,7.00132 L24.9999268,7.70847994 L7.70712678,24.99868 Z\\"></path></svg><span class=\\"screen-reader-text\\">actions.dismiss</span></button></div></div></div></div></div></section>"`;
